compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7031,6 +7045,27 @@ private BType checkFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr, BType
         return actualType;
     }
 
+    private boolean checkRecordFieldExistence(BRecordType recordVarRefType, Name fieldName) {
+        return recordVarRefType.fields.containsKey(fieldName.getValue());
```

Comment:
Do we need a method for this? IMO we can do this in-line since it's straightforward, self-explanatory, and the only place this is used.

Reply:
Removed with the new approach.